### PR TITLE
Allow get_task_auth (used by fcm make) to return None if

### PR DESCRIPTION
**Pull request by wxtim**
cylc.rose is not installed.

Proved horrendous to test - @dpmatthews  - can you test manually, whilst @oliver-sanders checks for Python goodness?

### CODE REVIEW

File: metomi/rose/suite_engine_procs/cylc.py [hunk 1]
```diff
@@ -107,7 +107,12 @@ def get_task_auth(
         from cylc.flow.exceptions import WorkflowFilesError
         from cylc.flow.hostuserutil import is_remote_platform
         from cylc.flow.platforms import get_host_from_platform
-        from cylc.rose.platform_utils import get_platform_from_task_def
+
+        try:
+            from cylc.rose.platform_utils import get_platform_from_task_def
+        except ModuleNotFoundError:
+            return None
```
Comment (dpmatthews) [inline metomi/rose/suite_engine_procs/cylc.py:114]: Maybe add a comment to explain, e.g.
```
# Allow single stage fcm_make app to work without requiring cylc.rose
```

**Comment by dpmatthews** [comment]
Works fine in my tests

**Comment by oliver-sanders** [comment]
The solution is fine, however, it has the consequence that the interface will always return `None` if cylc-rose is not installed. What are the consequences of this?

**Comment by dpmatthews** [comment]
> The solution is fine, however, it has the consequence that the interface will always return `None` if cylc-rose is not installed. What are the consequences of this?

This function is only used by the fcm_make built-in app. Returning `None` is equivalent to there being no fcm_make2 task found or no workflow file found which is fine - 2 stage fcm_make is only supported on the localhost install target (because the workflow files aren't mirrored).

File: metomi/rose/suite_engine_procs/cylc.py [hunk 2]
```diff
@@ -107,7 +107,13 @@ def get_task_auth(
         from cylc.flow.exceptions import WorkflowFilesError
         from cylc.flow.hostuserutil import is_remote_platform
         from cylc.flow.platforms import get_host_from_platform
-        from cylc.rose.platform_utils import get_platform_from_task_def
+        try:
+            from cylc.rose.platform_utils import get_platform_from_task_def
+        except:
```
Comment (wxtim) [inline metomi/rose/suite_engine_procs/cylc.py:112]:
```suggestion
        except ModuleImportError:
```

**Review by dpmatthews — APPROVED** [review]

**Comment by oliver-sanders** [comment]
Closes https://github.com/metomi/rose/issues/2575 ?

**Review by oliver-sanders — APPROVED** [review]

**Comment by hjoliver** [comment]
> +1 @wxtim could you copy this comment into the docstring, would be good to capture this. The returns section of the docstring could do with an extra bullet point to match.

I've pushed a commit to do this, so I can merge for the forthcoming release...

**Pull request merged**